docs/en-us/dev/user_doc/alert_spi.md

```diff
@@ -0,0 +1,102 @@
+### DolphinScheduler Alert SPI main design
+
+#### DolphinScheduler SPI Design
+
+DolphinScheduler is in the middle of a microkernel + plug-in architecture change. All core capabilities such as tasks, resource storage, registry etc. will be designed to be extensions, and we want to improve the flexibility as well as the friendliness (extensibility) of DolphinScheduler itself through SPI.
+
+You can read the relevant code under the dolphinscheduler-spi module as a reference. The extended interface of the associated plug-in is under the module, and when we need to implement the relevant function plug-in, it is recommended to read this block first. Of course, you are advised to read the documentation, which will take a lot less time.  However, the documentation has a certain lag and when it is missing, it is suggested that the source code prevails (If you are interested, we also welcome you to submit relevant documents).  In addition, we make few changes (excluding additions) to the extension interface, except for major architectural changes where there is an incompatible upgrade version, so existing documentation is generally adequate.

```

Comment:
block -> code block
advised -> welcomed
take a lot less time -> save a lot of time
Two spaces are before "however".
 it is suggested that the source code prevails => it is suggested to take the source code as the reference.
Two spaces are before "In addition".
make few changes to -> rarely change the extension 
, so existing documentation is generally adequate. => Hence, the existing documentation is generally adequate.